src/ir/datalog/operation_test.cc

```diff
@@ -0,0 +1,66 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/ir/datalog/operation.h"
+
+#include "absl/strings/string_view.h"
+#include "src/common/testing/gtest.h"
+#include "src/ir/datalog/input_relation_fact.h"
+#include "src/ir/datalog/value.h"
+
+namespace raksha::ir::datalog {
+
+using testing::Combine;
+using testing::TestWithParam;
+using testing::ValuesIn;
+
+struct IsOperationFactAndExpectedDatalog {
+  const IsOperationFact *is_operation_fact;
+  absl::string_view expected_datalog;
+};
+
+class IsOperationFactTest
+    : public TestWithParam<IsOperationFactAndExpectedDatalog> {};
+
+TEST_P(IsOperationFactTest, IsOperationFactTest) {
+  auto [is_operation_fact, expected_datalog] = GetParam();
+  EXPECT_EQ(is_operation_fact->ToDatalogString(), expected_datalog);
+}
+
+static const IsOperationFact kSampleFact1 = IsOperationFact(Operation(
```

Comment:
Will using `intrusive_ptr` enable us  to create these values directly in `INSTANTIATE_TEST_SUITE_P`? (If so, OK to defer for later.)

Reply:
We'd have to use `intrusive_ptr` throughout the Souffle IR data structures, but that would allow us to use them directly, yes.

Reply:
It would be pretty nice to be able to copy these files, but using a more exotic, bespoke data structure (like `intrusive_ptr`) may affect our ability to export this code to other projects (as we briefly discussed). Deferring for now.